### PR TITLE
[BACKLOG-15946] Multiple changes.

### PR DESCRIPTION
**Pull request by dcleao**
* Removing spaces from some folder names to workaround the requirejs-maven-plugin bug (BACKLOG-16501)
* Removing javascript folder from resources as it is now being properly handled by the bundle-javascript profile
* Updating VizAPI sample files

@pamval, @dkincade please review.
/cc @pentaho/millenniumfalcon

### CODE REVIEW

File: Samples_for_Extending_Pentaho/javascript-apis/platform/pentaho/visual/samples/bar-d3-sandbox/README.md [hunk 1]
```diff
@@ -5,4 +5,4 @@ This folder contains the Bar/D3 visualization sample, at the sandbox stage.
 To quickly view the visualization in operation, execute `npm install` in this folder and open `index.html` through 
 a local web-server (such as Python's, `python -m SimpleHTTPServer 8000`).
 
-For more information, see `http://javascript-apis.pentaho.com/platform/pentaho/visual/samples/bar-d3-sandbox/`.
+For more information, see http://javascript-apis.pentaho.com/platform/pentaho/visual/samples/bar-d3-sandbox/.
```
Comment (nantunes) [inline Samples_for_Extending_Pentaho/javascript-apis/platform/pentaho/visual/samples/bar-d3-sandbox/README.md]: This URL is now wrong.

File: Samples_for_Extending_Pentaho/javascript-apis/platform/pentaho/visual/samples/bar-d3-sandbox/package.json [hunk 2]
```diff
@@ -5,7 +5,9 @@
   "description": "Pentaho Visual Samples D3-Bar",
   "license": "MIT",
   "dependencies": {
-    "@nantunes/viz-api": "https://github.com/nantunes/pentaho-viz-sandbox/releases/download/3.0.4/nantunes-viz-api-3.0.4.tgz",
     "d3": "^4.8.0"
-  }
+  },
+  "devDependencies": {
+    "@nantunes/viz-api": "https://github.com/nantunes/pentaho-viz-sandbox/releases/download/3.0.4/nantunes-viz-api-3.0.4.tgz"
```
Comment (nantunes) [inline Samples_for_Extending_Pentaho/javascript-apis/platform/pentaho/visual/samples/bar-d3-sandbox/package.json]: This must change. It is ok to generate the VizAPI dev release package?

Comment (dcleao) [reply Samples_for_Extending_Pentaho/javascript-apis/platform/pentaho/visual/samples/bar-d3-sandbox/package.json]: Do you have an URL and final package name?

Comment (nantunes) [reply Samples_for_Extending_Pentaho/javascript-apis/platform/pentaho/visual/samples/bar-d3-sandbox/package.json]: Not yet.

**Comment by dcleao** [comment]
Fixed README URLs to sandbox sample.

**Comment by dcleao** [comment]
This is now ready for merging.
@pamval, @dkincade please review.

**Comment by dcleao** [comment]
Ping @pamval, @dkincade.

**Comment by dcleao** [comment]
Ping @pamval, @dkincade.

**Pull request merged**